src/sourceItemWrapper.js

```diff
@@ -489,20 +489,92 @@ class SourceItemWrapper extends ItemWrapper {
+        translation.setString(str);
+
+        const progress = new Progress(
+            'loading',
+            'Adding citations...'
         );
+        
+        try {
+            var translators = await translation.getTranslators();
+
+            if (translators.length > 0){
+                // set libraryID to false so we don't save this item in the Zotero library
+                var newItems = await translation.translate({libraryID: false});
+        
+                let citations = []
+                if (newItems.length > 0){
+                    newItems.forEach((item) => {
+                        let newItem = new Zotero.Item(item.itemType);
+                        newItem.fromJSON(item);
+            
+                        let citation = new Citation({item: newItem, ocis: []}, this);
+                        citation.target.item = newItem;
+                        citations.push(citation)
+                    });
+                    
+                    this.addCitations(citations);
+        
+                    progress.updateLine(
+                        'done',
+                        'Added citations'
+                    );
+
+                    return;
+                }
+            }
+            // no translators, or no new items were detected
+            progress.updateLine(
+                'error',
+                'No citations were detected'
+            );  
+        }
+        catch {
+            progress.updateLine(
+                'error',
+                'Error adding citations'
+            );
+        }
+
     }
 
     exportToBibTeX(citationIndex) {
-        Services.prompt.alert(
-            window,
-            Wikicite.getString('wikicite.global.unsupported'),
-            Wikicite.getString('wikicite.bibtex.export-citations.unsupported')
-        );
+        if (this._citations.length) {
+            this.loadCitations();
+            var exporter = new Zotero_File_Exporter();
+
+            // export all citations, or only those selected?
+            var citationsToExport;
+            if (citationIndex === undefined){
+                citationsToExport = this._citations;
+            }
+            else{
+                citationsToExport = [this._citations[citationIndex]];
+            }
+
+            // extract the Zotero items from the citations
+            var citationItems = citationsToExport.map((citation) => {
+                // Note: need to set the libraryID for the exported items, otherwise we get an error on export
+                citation.target.item._libraryID = this.item._libraryID;
+                return citation.target.item;
+            });
+            
+            exporter.items = citationItems;
+            if(!exporter.items || !exporter.items.length) throw("no citations to export");
+            
+            // opens Zotero export dialog box - can select format and file location
+            exporter.save();
```

Comment:
What about adding `exporter.name = Wikicite.getString(`wikicite.item-menu.export-file.filename');` so the default file name is something like "Exported citations" instead of "Exported items"?